pkg/cmd/pipelineresource/list.go

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipelineresource
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/printer"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"text/tabwriter"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cliopts "k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const (
+	msgNoPREsFound = "No pipelineresources found."
+)
+
+func listCommand(p cli.Params) *cobra.Command {
+	f := cliopts.NewPrintFlags("list")
+	eg := `
+# List all PipelineResources in a namespaces 'foo'
+tkn pre list -n foo",
+`
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List pipeline resources in a namespace",
+		Example: eg,
+		SilenceUsage: true,
+		Args:         cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			pres, err := list(p)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to list pipelineresources from %s namespace \n", p.Namespace())
+				return err
+			}
+
+			output, err := cmd.LocalFlags().GetString("output")
+			if err != nil {
+				fmt.Fprint(os.Stderr, "Error: output option not set properly \n")
+				return err
+			}
+
+			if output != "" {
+				return printer.PrintObject(cmd.OutOrStdout(), pres, f)
+			}
+
+			err = printFormatted(cmd.OutOrStdout(), pres)
+			if err != nil {
+				fmt.Fprint(os.Stderr, "Failed to print Pipelineresources \n")
+				return err
+			}
+			return nil
+		},
+
+	}
+
+	f.AddFlags(cmd)
+
+	return cmd
+}
+
+func list(p cli.Params) (*v1alpha1.PipelineResourceList, error) {
+	cs, err := p.Clients()
+	if err != nil {
+		return nil, err
+	}
+
+	options := v1.ListOptions{}
+	prec := cs.Tekton.TektonV1alpha1().PipelineResources(p.Namespace())
+	pres, err := prec.List(options)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE: this is required for -o json|yaml to work properly since
+	// tektoncd go client fails to set these; probably a bug
+	pres.GetObjectKind().SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Version: "tekton.dev/v1alpha1",
+			Kind:    "PipelineRunList",
+		})
+
+	return pres, nil
+}
+
+func printFormatted(out io.Writer, pres *v1alpha1.PipelineResourceList) error {
+	if len(pres.Items) == 0 {
+		fmt.Fprintln(out, msgNoPREsFound)
```

Comment:
could you please use `stderr` here please? perhaps the method should receive `cli.Stream{Out, Err}` instead of just `out`